session.go

```diff
@@ -260,9 +260,8 @@ func (s *Session) KeyspaceMetadata(keyspace string) (*KeyspaceMetadata, error) {
 // returns routing key indexes and type info
 func (s *Session) routingKeyInfo(stmt string) (*routingKeyInfo, error) {
 	s.routingKeyInfoCache.mu.Lock()
-	cacheKey := s.cfg.Keyspace + stmt
```

Comment:
im not sure this is a great idea, even though the cache uses the config keyspace which is static, it should really use the keyspace from the connection pool which is a bug. Can you remove this change and check the performance again?